src/sentry/relay/config/metric_extraction.py

```diff
@@ -290,10 +290,12 @@ def _convert_snuba_query_to_metrics(
     return _convert_aggregate_and_query_to_metrics(
         project,
         snuba_query.dataset,
-        snuba_query.aggregate,
```

Comment:
Most of the changes in this PR are to simplify the signature of `_convert_aggregate_and_query_to_metrics`.
Type checking will happen as part of calling OnDemandMetricSpec
<img width="516" alt="image" src="https://github.com/getsentry/sentry/assets/44410/72f5b9cf-38bd-42d9-8ca8-24ebb4e99055">